frontend/src/pages/Assessments/Quiz.tsx

```diff
@@ -57,8 +57,8 @@ const Quiz = ({
   }
 
   useEffect(() => {
-    if (!assessment) getAssessment(params.id);
-  }, [assessment, getAssessment, params]);
+    if (!assessment && !isFinished) getAssessment(params.id);
```

Comment:
i think this was the root of the problem, right after assessments would be submitted there was a chance that this useEffect would re-run because the assessment would be set to undefined, but then getAssessment would reload in the same assessment - meaning that when the user would come back for new assessment it would read in the old assessment. now we don't reload if the isFinished flag is true. If this doesn't fix it my other potential fix is to cross-check the `params.id` with the `assessmentId` cause we don't want those to be out of sync

Reply:
dillon you're a legend
i'll look at this tomorrow (but lowkey any better ideas of testing besides "just run through assessments and stuff a lot"